### ISSUE TITLE
Quietness issues

### ISSUE THREAD

**Issue by Robinlovelace** [nptscot/npt#66]
Good example of where quietness scores are way out below. Heads-up @mvl22 please take a look:

![image](https://user-images.githubusercontent.com/1825120/224509705-6a858ed7-e072-409f-9d60-0bba8cedb9af.png)

Source: https://nptscot.github.io/#14.01/55.95428/-3.2096

Road with quietness of 40: https://www.google.com/maps/@55.9576566,-3.2309974,3a,75y,240.16h,67.73t/data=!3m6!1e1!3m4!1sDpOCJePwhxD7BTW9CsVADA!2e0!7i16384!8i8192

Road with quietness of 81 that I walked on with @mem48 ealier this week: https://www.google.com/maps/@55.951296,-3.200337,3a,75y,82.33h,76.97t/data=!3m6!1e1!3m4!1sAF1QipM9lxoDRK3T39AJ4C4WTQdIf9sFvBVQSTN60HHl!2e10!7i11000!8i5500

Not sure how best to deal with this: post-processing fix or can it be fixed upstream, e.g. shared use busway and lots of bus stops -> not quiet?

**Comment by mvl22** [comment]
> Road with quietness of 40:
> https://www.google.com/maps/@55.9576566,-3.2309974,3a,75y,240.16h,67.73t/data=!3m6!1e1!3m4!1sDpOCJePwhxD7BTW9CsVADA!2e0!7i16384!8i8192

That's not a cycle lane. It's a B road with car parking. I really can't imagine a parent with child cycling along there, can you?

https://www.google.com/maps/@55.9576568,-3.2309452,3a,75y,254.84h,74.38t/data=!3m7!1e1!3m5!1sj-61IPCEqOBgRnViK-Jm2Q!2e0!5s20180501T000000!7i16384!8i8192

**Comment by mvl22** [comment]
> Road with quietness of 81 that I walked on with @mem48 ealier this week:
> https://www.google.com/maps/@55.951296,-3.200337,3a,75y,82.33h,76.97t/data=!3m6!1e1!3m4!1sAF1QipM9lxoDRK3T39AJ4C4WTQdIf9sFvBVQSTN60HHl!2e10!7i11000!8i5500

The issue here is trams support, which we need to sort out. In general trams have not been a high priority for implementation as there's so little infrastructure in the UK. In practice the infrastructure in central areas with a tram line is going to be pretty location-specific. Key issues are whether the amount of space aside from the tram lines themselves is sufficiently wide, and other traffic taking up the space. Princes Street I agree in my experience isn't that nice to cycle on, though arguably I'd say it's better than many similar roads in city centres where cars are permitted.

**Comment by Robinlovelace** [comment]
> That's not a cycle lane. It's a B road with car parking. I really can't imagine a parent with child cycling along there, can you? I'd say 40 is very slightly on the low side but not that far off.

I can and would. The point is the relative quietness and cyclability for people who are not confident. I'm confident most people would want to cycle alongside a child in this example, with a CycleStreets quietness of 40, more than on Princes Street.

With Princes Street it's not just the trams, it's also all the fast moving double deckers and other heavy traffic going up and down. As you say, Princes Street is clearly a busy (not to mention scary) central road with heavy traffic that should not have a quietness score of 80. The question is how to tweak the score based on OSM tags, not sure of the best solution there.

**Comment by mem48** [comment]
@mvl22 A thought on the bus/tram issue is that https://github.com/creds2/CarbonCalculator/releases/download/1.0/PBCC_transit_stop_frequency_2020.zip contains a measure of service frequency for all public transport stops in GB. Perhaps we could tag up roads with a frequent service and reduce the quietness a bit. Not sure how you would detect the roads between bus stops though. I haven't used it but https://github.com/ad-freiburg/pfaedle might be a solution.

**Comment by mvl22** [comment]
We've discussed the tram case and the quietness will be adjusted. As I say, this issue is rather location-specific. @si-the-pie will reply when done.

**Comment by Robinlovelace** [comment]
Another issue where we have the opposite problem, a road with light segregation that is down as very unpleasant to cycle, streetview:

![image](https://user-images.githubusercontent.com/1825120/225704734-e9755a6f-2253-4d58-8427-6aa9246ef47a.png)

Here it is in our tool: https://www.npt.scot/#16.57/55.931991/-3.172892/-50.4 

Any ideas of the cause, a problem with OSM tagging I guess, although I've heard OSM is good in Edinburgh so not sure.

**Comment by si-the-pie** [comment]
The Princes Street, Edinburgh tram sections have now been hacked so that the ways along there have quietness of 40%. This should be live by Friday afternoon.
This is a temporary measure until support for roads with trams is available.

**Comment by si-the-pie** [comment]
Mayfield Gardens, A701 has lanes=3 and so is capped at quietness 20%.

**Comment by joeytalbot** [comment]
This is the example I was thinking of. It's one that I told you about previously, where a cycle path on the pavement of an 8 lane highway has a high quietness rating. The quietness rating was originally 85%, but you reduced it after our conversation and is now 42%. 

https://www.cyclestreets.net/journey/88843644/

**Comment by si-the-pie** [comment]
Good point about Mayfield Gardens where the light segregation offers some protection.

The rule that picks up `lanes > 2` now avoids ways that have `cycleway=track`. 

Most of that row now comes out at 40% apart from https://www.openstreetmap.org/way/3993998 where the busway is shared, and is still 20%. This will be live on Saturday afternoon on the main site.

**Comment by mvl22** [comment]
> Could you please revisit this rule in cases where there is separation between the cycleway and motor traffic, in this case light segregation with wands?

Yes, the issue here is the relatively unusual cases where cycleway=track is added as part of the road itself. It's sufficiently rare it's not been high up enough on my list to look at. There is a bit of a holy war in OSM as to whether these should be part of the road or separately-drawn, mainly because this kind of segregation is highly ambiguous as to whether it is part of the road or not. The new [separation=*](https://wiki.openstreetmap.org/wiki/Key:separation) standard should start to resolve this.

PS I think the OSM data may be wrong here anyway - I suspect the lanes count is wrong following the addition of the cycleways. I contacted the person that updated it.

**Comment by si-the-pie** [comment]
They [cycleway by the Lucan bypass](https://www.openstreetmap.org/way/945978380/
) Is scored as quietness 50% because it has `width=1`.

The bit of that path that is used by [the route Joey mentioned ](https://www.cyclestreets.net/journey/88843644/)came out as 42% because of the hill climbing effect.

**Comment by Robinlovelace** [comment]
Thanks for the updates @mvl22 and @si-the-pie. I think those changes will fix the issue, will keep this issue open and close when we have the update route networks in the app at www.npt.scot.

**Comment by Robinlovelace** [comment]
Good news, these are fixed in the latest route networks uploaded to our releases, many thanks @mvl22 and team!

Illustration below for Princes Street in context.

![image](https://user-images.githubusercontent.com/1825120/227735766-9b0bef0a-3658-4c3b-9841-cb77f753678e.png)

This is fixed in terms of data but we need to update the production website before we can close this issue. Assigning to @mem48 accordingly.

**Comment by Robinlovelace** [comment]
Update @mem48: https://github.com/nptscot/outputdata/releases/download/v2023-03-28-15-53-50_commit_e59db4597b54b75d29407a61a326436fa7402c60/rnet.pmtiles

**Comment by Robinlovelace** [comment]
See .zip file here @mem48 https://github.com/nptscot/outputdata/releases/tag/v2023-03-28-15-53-50_commit_e59db4597b54b75d29407a61a326436fa7402c60

**Comment by Robinlovelace** [comment]
New route network is quite different and I think better in many ways, after https://github.com/nptscot/nptscot.github.io/pull/56 is merged.

Before

![image](https://user-images.githubusercontent.com/1825120/229315284-b6400486-9134-4cde-9391-6a82ebfe6ffc.png)

After

![image](https://user-images.githubusercontent.com/1825120/229315302-81ba55ad-4dd1-4169-bee2-46bbc3827f5e.png)

**Comment by Robinlovelace** [comment]
Importantly the new quietness scores are up, you lovely new data will be on www.npt.scot soon @mvl22 thanks for the build! Can you let us know when the next build based on the latest OSM data is scheduled?

![image](https://user-images.githubusercontent.com/1825120/229315369-1496e21e-05dd-458d-a6b6-0031bd945da1.png)

We can close this issue when the new network data is deployed on the production branch and visible to users on npt.scot.

**Comment by Robinlovelace** [comment]
Fixed on production also, great work Malcolm on getting the tiles up there!

![image](https://user-images.githubusercontent.com/1825120/229346111-e4574e38-c980-4fa4-bab8-393cc8cd1ff6.png)

Very happy about this.
